integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -203,9 +203,65 @@ public void shouldModifyProduceMessage(KafkaCluster cluster, Admin admin) throws
         }
     }
 
+    @Test
+    public void offsetManuallyManaged(KafkaCluster cluster, Admin admin) throws Exception {
+        admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+
+        String config = """
+                proxy:
+                  address: %s
+                clusters:
+                  demo:
+                    bootstrap_servers: %s
+                filters:
+                - type: ApiVersions
+                - type: BrokerAddress
+                """.formatted(PROXY_ADDRESS, cluster.getBootstrapServers());
+
+        try (var proxy = startProxy(config)) {
+            try (var producer = new KafkaProducer<String, String>(Map.of(
+                    ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, PROXY_ADDRESS,
+                    ProducerConfig.CLIENT_ID_CONFIG, "offsetManuallyManaged",
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000))) {
+                producer.send(new ProducerRecord<>(TOPIC_1, "my-key", MESSAGE_TEXT)).get();
+                producer.send(new ProducerRecord<>(TOPIC_1, "my-key2", MESSAGE_TEXT)).get();
+                producer.flush();
+            }
+
+            ConsumerRecords<String, String> records1;
+            long offsetCommitted;
+            long actualEndOffset;
+            try (var consumer = new KafkaConsumer<String, String>(Map.of(
+                    ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, PROXY_ADDRESS,
+                    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.GROUP_ID_CONFIG, "my-group-id",
+                    ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false",
+                    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+
+                Set<TopicPartition> partitions = new HashSet<>();
+                TopicPartition actualTopicPartition = new TopicPartition(TOPIC_1, 0);
+                partitions.add(actualTopicPartition);
+                consumer.assign(Set.of(actualTopicPartition));
+
+                records1 = consumer.poll(Duration.ofSeconds(100));
+
+                consumer.commitSync();
+
+                var committed = consumer.committed(partitions);
+                OffsetAndMetadata offsetAndMetadata = committed.get(actualTopicPartition);
+                offsetCommitted = offsetAndMetadata.offset();
+                actualEndOffset = consumer.endOffsets(partitions).get(actualTopicPartition);
+            }
+            assertEquals(offsetCommitted, actualEndOffset);
```

Comment:
This is not correct, actually. You are verifying kafka is storing committed offset correctly via `committed` and `endOffsets` API, but not verifying we're committing the expected offsets. We can use this [API](https://javadoc.io/static/org.apache.kafka/kafka-clients/3.4.0/org/apache/kafka/clients/consumer/KafkaConsumer.html#commitSync-java.util.Map-) to commit our "expected offset". And then retrieve the "actual offset" from broker via `committed` API, to make sure kroxy is working as expected.

Reply:
Thanks @showuon! Changed using the suggested API. Please, could you review it again and see if I misunderstood anything?